### PR TITLE
feat(): Adding aliases to service export

### PR DESCRIPTION
**Pull request by bharath-avesha**
There is a requirement to identify exported services with arbitrary names instead of slice.local names. This can be achieved by providing an alias field in the service export spec section so that the users can give any name that they desire to the service being exported from a cluster.

Added a new field to hold alias names to the service import and export CRDs.

### CODE REVIEW

**Review by gourishkb — APPROVED** [review]

**Review by gourishkb — APPROVED** [review]

File: controllers/serviceexport/serviceexport.go [hunk 1]
```diff
@@ -229,3 +229,52 @@ func (r *Reconciler) SyncSvcExportStatus(ctx context.Context, serviceexport *kub
...
+			break
+		}
+	}
+
+	if updateNeeded {
+		serviceexport.Status.Aliases = serviceexport.Spec.Aliases
+		serviceexport.Status.LastSync = 0
+		err := r.Status().Update(ctx, serviceexport)
+		if err != nil {
+			log.Error(err, "Failed to update serviceexport ports")
```
Comment (Rahul-D78) [inline controllers/serviceexport/serviceexport.go:262]: Can we have an error message - `Failed to update serviceexport status with Aliases`

File: api/v1beta1/serviceimport_types.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ type ServiceImportSpec struct {
 	DNSName string `json:"dnsName"`
 	// Ports which should be exposed through the service
 	Ports []ServicePort `json:"ports"`
+	// Alias names for the exported service. The service could be addressed by the alias names
+	// in addition to the slice.local name.
+	Aliases []string `json:"aliases,omitempty"`
```
Comment (Rahul-D78) [inline api/v1beta1/serviceimport_types.go:49]: We don't have a function to reconcile Aliases for serviceimport. Where as `ReconcileAliases` function present for serviceexport.

**Comment by gourishkb** [comment]
Do we still need this PR ? @bharath-avesha shall we close it ?

**Comment by bharath-avesha** [comment]
@gourishkb, i have closed it.

**Pull request closed without merging**